processor/spanmetricsprocessor/processor.go

```diff
@@ -282,8 +296,24 @@ func (p *processorImp) collectCallMetrics(ilm pdata.InstrumentationLibraryMetric
 		dpCalls.SetTimestamp(pdata.NewTimestampFromTime(time.Now()))
 		dpCalls.SetIntVal(p.callSum[key])
 
-		p.metricKeyToDimensions[key].CopyTo(dpCalls.Attributes())
+		if dimensions, okay := p.getDimensionsByMetricKey(key); okay {
+			dimensions.CopyTo(dpCalls.Attributes())
+		}
+	}
+}
+
+// getDimensionsByMetricKey get dimensions from `metricKeyToDimensions` cache
```

Comment:
nitpick: Please end comments with full-stops.

```suggestion
// getDimensionsByMetricKey gets dimensions from `metricKeyToDimensions` cache.
```